src/accessorysetupkit.cs

```diff
@@ -0,0 +1,208 @@
+
+		[Export ("finishAuthorization:settings:completionHandler:")]
+		void FinishAuthorization (ASAccessory accessory, ASAccessorySettings settings, ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("removeAccessory:completionHandler:")]
+		void RemoveAccessory (ASAccessory accessory, ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("renameAccessory:options:completionHandler:")]
+		void RenameAccessory (ASAccessory accessory, ASAccessoryRenameOptions renameOptions, ASAccessorySessionCompletionHandler completionHandler);
+	}
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASAccessorySettings {
+		[Export ("SSID", ArgumentSemantic.Copy), NullAllowed]
+		string Ssid { get; set; }
+	}
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASDiscoveryDescriptor {
+		[Export ("supportedOptions", ArgumentSemantic.Assign)]
+		ASAccessorySupportOptions SupportedOptions { get; set; }
+
+		[Export ("bluetoothCompanyIdentifier", ArgumentSemantic.Assign)]
+		ushort /* ASBluetoothCompanyIdentifier */ BluetoothCompanyIdentifier { get; set; }
+
+		[Export ("bluetoothManufacturerDataBlob", ArgumentSemantic.Copy), NullAllowed]
+		NSData BluetoothManufacturerDataBlob { get; set; }
+
+		[Export ("bluetoothManufacturerDataMask", ArgumentSemantic.Copy), NullAllowed]
+		NSData BluetoothManufacturerDataMask { get; set; }
+
+		[Export ("bluetoothNameSubstring", ArgumentSemantic.Copy), NullAllowed]
+		string BluetoothNameSubstring { get; set; }
+
+		[Export ("bluetoothServiceDataBlob", ArgumentSemantic.Copy), NullAllowed]
+		NSData BluetoothServiceDataBlob { get; set; }
+
+		[Export ("bluetoothServiceDataMask", ArgumentSemantic.Copy), NullAllowed]
+		NSData BluetoothServiceDataMask { get; set; }
+
+		[Export ("bluetoothServiceUUID", ArgumentSemantic.Copy), NullAllowed]
+		CBUUID BluetoothServiceUuid { get; set; }
+		[Export ("SSID", ArgumentSemantic.Copy), NullAllowed]
+		string Ssid { get; set; }
+
+		[Export ("SSIDPrefix", ArgumentSemantic.Copy), NullAllowed]
+		string SsidPrefix { get; set; }
+	}
+
+	[Native]
+	[iOS (18, 0)]
+	[ErrorDomain ("ASErrorDomain")]
+	enum ASErrorCode : long {
+		Success             = 0,
+		Unknown             = 1,
+		ActivationFailed    = 100,
+		DiscoveryTimeout    = 200,
+		ExtensionNotFound   = 300,
+		Invalidated         = 400,
+		PickerAlreadyActive = 500,
+		PickerRestricted    = 550,
+		UserCancelled       = 700,
+		UserRestricted      = 750,
+	}
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASPickerDisplayItem {
+		[Export ("allowsRename", ArgumentSemantic.Assign)]
+		bool AllowsRename { get; set; }
+
+		[Export ("renameOptions", ArgumentSemantic.Assign)]
+		ASAccessoryRenameOptions RenameOptions { get; set; }
+
+		[Export ("name", ArgumentSemantic.Copy)]
+		NSString Name { get; set; }
```

Comment:
why `NSString` here but `.ctor` bellow is `string`?

Reply:
Fixed.